tests/L0/run_transformer/test_weight_grad.py

```diff
@@ -55,81 +40,72 @@ def backward_matrix_operation(self, alpha, beta, A, B, C):
+            #print(f"BATCH_SIZE = {BATCH_SIZE}, SEQUENCE_LENGTH = {SEQUENCE_LENGTH}, HIDDEN_SIZE = {HIDDEN_SIZE}, OUTPUT_SIZE = {OUTPUT_SIZE}")
             input_shape = (SEQUENCE_LENGTH, BATCH_SIZE, HIDDEN_SIZE) # k*m
             output_shape = (SEQUENCE_LENGTH, BATCH_SIZE, OUTPUT_SIZE) # k*n
             weight_shape = (OUTPUT_SIZE, HIDDEN_SIZE)
             
             torch.manual_seed(self.SEED)
 
             input = torch.randn(input_shape, requires_grad=False, device="cuda") # A
-            output = torch.randn(output_shape, requires_grad=False, device="cuda") #B
-            weight = torch.randn(weight_shape, requires_grad=False, device="cuda") # C
+            grad_output = torch.randn(output_shape, requires_grad=False, device="cuda") #B
+            grad_weight = torch.randn(weight_shape, requires_grad=False, device="cuda") # C
 
-            ref_weight = weight.clone()
+            ref_grad_weight = grad_weight.clone()
             
             if ab_type == 'fp16':
                 input = input.half()
-                output = output.half()
+                grad_output = grad_output.half()
             elif ab_type == 'bf16':
                 input = input.bfloat16()
-                output = output.bfloat16()
+                grad_output = grad_output.bfloat16()
         
             if cd_type == 'fp16':
-                weight = weight.half()
-                ref_weight = ref_weight.half()
+                grad_weight = grad_weight.half()
+                ref_grad_weight = ref_grad_weight.half()
             elif cd_type == 'bf16':
-                weight = weight.bfloat16()
-                ref_weight = ref_weight.bfloat16()
+                grad_weight = grad_weight.bfloat16()
+                ref_grad_weight = ref_grad_weight.bfloat16()
 
             if cd_type == 'fp32':
                 fused_weight_gradient_mlp_cuda.wgrad_gemm_accum_fp32(
-                    input, output, weight
+                    input, grad_output, grad_weight
                 )
             else: 
                 fused_weight_gradient_mlp_cuda.wgrad_gemm_accum_fp16(
-                    input, output, weight
+                    input, grad_output, grad_weight
                 )
-            
-            #ref linear
+
             alpha = 1.0
             beta = 1.0
-            self.backward_matrix_operation(alpha, beta, input, output, ref_weight)
-
-            #cosim similarity
-            weight = weight.view(-1)
-            ref_weight = ref_weight.view(-1)
+            self.backward_matrix_operation(alpha, beta, input, grad_output, ref_grad_weight)
 
-            #compare
-            similarity = self.cosin_similarity(weight, ref_weight)
-            print("similarity :", similarity)
-            if similarity > 0.999:
-                print("========PASS======")
-            else:
-                print("========FAIL======")
-               
+            grad_weight = grad_weight.view(-1)
```

Comment:
We don't need to reshape the tensor now as `torch.allclose` can take 2d tensors.